### PR TITLE
Add `mc-sgx-dcap-ql-types` crate

### PR DESCRIPTION
**Pull request by nick-mobilecoin**

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #141** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/141" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/141?utm_source=stack-comment).

**Review by awygle — APPROVED** [review]

File: dcap/ql/types/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+#![no_std]
+#![deny(missing_docs, missing_debug_implementations, unsafe_code)]
+
+use mc_sgx_core_types::FfiError;
+use mc_sgx_dcap_ql_sys_types::sgx_ql_path_type_t;
+
+/// Paths (location and filename) to be override the default entries.
+#[non_exhaustive]
+#[derive(Eq, PartialEq, Debug)]
+pub enum Path {
```
Comment (jcape) [inline dcap/ql/types/src/lib.rs:13]: Recommend calling this `PathKind` to disambiguate from `std::path::Path`

File: dcap/ql/types/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+[package]
+name = "mc-sgx-dcap-ql-typs"
```
Comment (jcape) [inline dcap/ql/types/Cargo.toml:2]:
```suggestion
name = "mc-sgx-dcap-ql-types"
```

**Review by jcape — CHANGES_REQUESTED** [review]
Clutch typo :-)

File: dcap/ql/types/src/lib.rs (hunk 1 above)
Comment (nick-mobilecoin) [reply dcap/ql/types/src/lib.rs]: Done

**Review by jcape — APPROVED** [review]

**Pull request merged**